### ISSUE TITLE
Support socks5 proxy

### ISSUE THREAD

**Issue by tiero** [grpc/grpc-node#1706]
### Is your feature request related to a problem? Please describe.
I want to use my tor client that expose a socks5 proxy, to connect to gRPC services exposed trough .onion endpoint

### Describe the solution you'd like

Add additional env var [here](https://github.com/grpc/grpc-node/blob/%40grpc/grpc-js%400.7.0/packages/grpc-js/src/http_proxy.ts#L45) something like `socks5_proxy`

### Describe alternatives you've considered
Using a client written in another language. ie. `grpc-go` has a custom net dialer you can pass down, so you can customize this.

**Comment by murgatroid99** [comment]
I want to note a couple of things. First, the library uses `http_proxy` and `grpc_proxy` interchangeably, with the only difference being that `grpc_proxy` is a higher priority, so adding another environment variable wouldn't change anything. And second, only the `http://` protocol is supported, so I would expect an error to be output if you use `socks5://` or `socks5h://`. So, the only way you're going to connect to gRPC services through TOR is by exposing a HTTP CONNECT proxy in your TOR client.

**Comment by tiero** [comment]
> so I would expect an error to be output if you use socks5:// or socks5h://.

Yes indeed, I got errors. Just mentioned for sake of completeness.

>  So, the only way you're going to connect to gRPC services through TOR is by exposing a HTTP CONNECT proxy in your TOR client.

Ok, so practically, I would need to add `HTTPTunnelPort` in my torrc file, is that correct?


PS: Any reason why we wouldn't directly support socks5 proxies? I see some setup where is tricky to change tor client configuration (ie. mobile evironment with orbot?)

Thanks

**Comment by murgatroid99** [comment]
I am not familiar with TOR configuration; you'll have to figure that part out for yourself.

My understanding is that grpc-go and other libraries do not actually directly support socks5 proxies. Rather, grpc-go provides an interface to use a custom dialer that can connect through a socks5 proxies, and the other libraries do not even provide that.

**Comment by tiero** [comment]
> grpc-go provides an interface to use a custom dialer that can connect through a socks5 proxies, and the other libraries do not even provide that.

Yes exaclty, I pass down a custom dialer in go. So yes, it's not that simple more how nodejs works vs go inner interfaces mechanics. Thanks, will try to circumnvent this via my tor client.

**Comment by tiero** [comment]
> What are the advantages of enabling a socks5 proxy as opposed to using the http proxy?

This means "tinkering" with the tor client configuration. 

I have use case in an app (wich uses grpc-js) where I need users to call an onion hidden service that exposes a grpc sever. If grpc-node only support http-proxy I cannot just say to the users "Hey open a Tor Browser and that's it" now you can use the app (wich is exposing a socks5 proxy on 9*50 port), but rather I need him to do extra instructions.

In go instead I can pass a custom dialer, so I can handle this out of grpc-go hands of course, in grpc-node instead we have this to be done inside.

Not a big deal, just nice to have it.
